### PR TITLE
Fix a few places where release name and namespace were conflated

### PR DESCRIPTION
**Pull request by sjmiller609**
wrt https://github.com/astronomer/issues/issues/1607

### CODE REVIEW

**Comment by sjmiller609** [comment]
I'm considering for a few minutes how we might use the unit testing for this situation in the future, but I'm not sure if it works well for not explicting calling out specific resources, i.e. some logic like "nothing should have namespace: {{ .Release.Name }}"

**Comment by sjmiller609** [comment]
I think the issue with the unit testing is that it does not play nicely to match many template files at once. It assumes that your assertion is only intended to apply to one template.

**Comment by danielhoherd** [comment]
> What are we doing elsewhere for these metadata namespace fields? I wonder if we should remove them altogether. Wouldn't everything automatically be set to the right namespace?

Seems to me like this would be a good approach except where an alternate namespace is explicitly needed.

**Comment by danielhoherd** [comment]
> I wonder if we should remove them altogether. Wouldn't everything automatically be set to the right namespace?

https://github.com/helm/helm/issues/5465 Says to not include them.

> In general, templates should not define a namespace. This is because Helm installs objects into the namespace provided with the --namespace flag.

**Comment by schnie** [comment]
Cool, seems like we should just rip them out then?

**Comment by sjmiller609** [comment]
> Cool, seems like we should just rip them out then?

Can we make that a separate github issue?

**Comment by schnie** [comment]
I'm just talking about the ones that are changed in this PR. Can we just remove these, rather than changing them? The rest of the charts should already be OK and not have these specified.

**Comment by sjmiller609** [comment]
> I'm just talking about the ones that are changed in this PR. Can we just remove these, rather than changing them? The rest of the charts should already be OK and not have these specified.

ok sgtm, thanks

File: charts/stan/templates/statefulset.yaml [hunk 1]
```diff
@@ -66,14 +66,14 @@ spec:
             - "dockerize"
           args:
             - "-wait"
-            - "tcp://astronomer-nats:4222"
+            - "tcp://{{ .Release.Name }}-nats:4222"
```
Comment (sjmiller609) [inline charts/stan/templates/statefulset.yaml:69]: also noticed this spot was hardcoded

**Comment by schnie** [comment]
LGTM

**Comment by sjmiller609** [comment]
> LGTM

thank you, i'll merge if CI passes

**Comment by sjmiller609** [comment]
picked into 0.16 https://github.com/astronomer/astronomer/commit/67cf7efcdc04c529663d12b025167d90dad64d7d

**Pull request merged**